### ISSUE TITLE
[Bug] Classifier with BinaryCrossentropy loss not learning

### ISSUE THREAD

**Issue by charlielito** [poets-ai/elegy#160]
**Describe the bug**
Training a linear classifier with elegy.losses.BinaryCrossentropy does not converge.

**Minimal code to reproduce**
Small snippet that contains a minimal amount of code.
```python
import os
from datetime import datetime

import dataget
from flax import linen
import elegy
import jax.numpy as jnp
import optax
import typer


def main(
    debug: bool = False,
    eager: bool = False,
    logdir: str = "runs",
    steps_per_epoch: int = 200,
    epochs: int = 100,
    batch_size: int = 64,
):

    if debug:
        import debugpy

        print("Waiting for debugger...")
        debugpy.listen(5678)
        debugpy.wait_for_client()

    current_time = datetime.now().strftime("%b%d_%H-%M-%S")
    logdir = os.path.join(logdir, current_time)

    X_train, y_train, X_test, y_test = dataget.image.mnist(global_cache=True).get()

    X_train = X_train[y_train <= 1]
    y_train = y_train[y_train <= 1]
    X_test = X_test[y_test <= 1]
    y_test = y_test[y_test <= 1]

    print("X_train:", X_train.shape, X_train.dtype)
    print("y_train:", y_train.shape, y_train.dtype)
    print("X_test:", X_test.shape, X_test.dtype)
    print("y_test:", y_test.shape, y_test.dtype)

    # import matplotlib.pyplot as plt

    # for i in range(10):
    #     plt.imshow(X_test[i, ...])
    #     plt.title(str(y_test[i]))
    #     plt.show()

    # return

    def accuracy(y_true, y_pred):
        return jnp.mean(jnp.argmax(y_pred, axis=-1) == y_true)

    class LinearClassifier(linen.Module):
        @linen.compact
        def __call__(self, x):
            x = jnp.reshape(x, (x.shape[0], -1)) / 255.0
            x = linen.Dense(1)(x)
            return x

    model = elegy.Model(
        module=LinearClassifier(),
        loss=elegy.losses.BinaryCrossentropy(from_logits=True),
        metrics=accuracy,
        optimizer=optax.adam(1e-3),
        run_eagerly=eager,
    )

    history = model.fit(
        x=X_train,
        y=y_train,
        epochs=epochs,
        steps_per_epoch=steps_per_epoch,
        batch_size=batch_size,
        validation_data=(X_test, y_test),
        shuffle=True,
        callbacks=[elegy.callbacks.TensorBoard(logdir=logdir)],
    )

    elegy.utils.plot_history(history)


if __name__ == "__main__":
    typer.run(main)

```

**Library Info**
0.4.1

**Comment by cgarciae** [comment]
@charlielito BCE is learning as you can see in the plot. What is happening here is that the definition of accuracy for multi-class should be:
```python
def accuracy(y_true, y_pred):
    y_true, y_pred = elegy.utils.maybe_expand_dims(y_true, y_pred)
    return jnp.mean((y_pred > 0) == y_true)
```
This is consistent with our definition of `elegy.metrics.BinaryAccuracy`.
